tests/modules/redshift_datasets/conftest.py

```diff
@@ -0,0 +1,201 @@
+import os
+import pytest
+
+from dataall.base.context import set_context, dispose_context, RequestContext
+
+from dataall.modules.redshift_datasets.services.redshift_connection_service import RedshiftConnectionService
+from dataall.modules.redshift_datasets.services.redshift_dataset_service import RedshiftDatasetService
+
+ENVNAME = os.environ.get('envname', 'pytest')
+
+
+class MockRedshiftDataClient:
+    def get_redshift_connection_database(self, *args, **kwargs):
+        return True
+
+    def list_redshift_schemas(self, *args, **kwargs):
+        return ['public', 'dev']
+
+    def list_redshift_tables(self, *args, **kwargs):
+        return [
+            {'name': 'table1', 'type': 'TABLE'},
+            {'name': 'table2', 'type': 'TABLE'},
+            {'name': 'table3', 'type': 'TABLE'},
+            {'name': 'table4', 'type': 'TABLE'},
+        ]
+
+    def list_redshift_table_columns(self, *args, **kwargs):
+        return [
+            {'name': 'column1', 'type': 'VARCHAR', 'nullable': True},
+            {'name': 'column2', 'type': 'INTEGER', 'nullable': False},
+            {'name': 'column3', 'type': 'DOUBLE', 'nullable': True},
+            {'name': 'column4', 'type': 'BOOLEAN', 'nullable': False},
+        ]
+
+
+class MockRedshiftClient:
+    def describe_cluster(self, *args, **kwargs):
+        return {'ClusterIdentifier': 'cluster_id_1', 'ClusterStatus': 'available'}
+
+
+class MockRedshiftServerlessClient:
+    def get_namespace_by_id(self, *args, **kwargs):
+        return {'namespaceId': 'XXXXXXXXXXXXXX', 'namespaceName': 'namespace_name_1'}
+
+    def list_workgroups_in_namespace(self, *args, **kwargs):
+        return [
+            {
+                'workgroupName': 'workgroup_name_1',
+                'workgroupArn': 'arn:aws:redshift-serverless:eu-west-1:XXXXXXXXXXXXXX:workgroup/workgroup_name_1',
+            }
+        ]
+
+    def get_workgroup_arn(self, *args, **kwargs):
+        return 'arn:aws:redshift-serverless:eu-west-1:XXXXXXXXXXXXXX:workgroup/workgroup_name_1'
+
+
+@pytest.fixture(scope='function')
+def patch_redshift(module_mocker):
```

Comment:
I think I disagree with creating Mock clients like that and I think there are practical dangers.
For example, let's say that you change your code and remove the `get_namespace_by_id` but you forget to remove the invocation from one line. Then you run the tests and they pass, you happily commit that to prod and then things start breaking because the function is missing in the real client.

I think the way to go here is...

```
fooClassMock = module_mocker.patch('path.to.foo.FooClass', autospec=True) # this returns a MagicMock bound to the FooClass 
fooClassMock.methoda.return_value = '123'
```

with `autospec=True` you make sure that calls are verified against the real class so if your code tries to access something that doesn't exist it will blow up.
Also this will allow you to assert that the expected methods were called with the correct arguments (if you have to validate that in any of your tests).